sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaPruningSuite.scala

```diff
@@ -155,6 +161,47 @@ class ParquetSchemaPruningSuite
       Row(null) :: Row(null) :: Nil)
   }
 
+  testSchemaPruning("select a single complex field and in where clause") {
+    val query1 = sql("select name.first from contacts where name.first = 'Jane'")
+    checkScan(query1, "struct<name:struct<first:string>>")
+    checkAnswer(query1, Row("Jane") :: Nil)
+
+    val query2 = sql("select name.first, name.last from contacts where name.first = 'Jane'")
+    checkScan(query2, "struct<name:struct<first:string,last:string>>")
+    checkAnswer(query2, Row("Jane", "Doe") :: Nil)
+
+    val query3 = sql("select name.first from contacts " +
+      "where employer.company.name = 'abc' and p = 1")
```

Comment:
Let's say a user adds  `where employer.company is not null`, can we still read schema with `employer:struct<company:struct<name:string>>>` as we only mark `contentAccessed = false` when `IsNotNull` is on an attribute?

Reply:
Added one query test for this case. Thanks.

Reply:
When there is a nested field access in the query like `employer.company.name`, then we don't need other fields inside `employ.company` other than `name`.

But if there is no such access but just `employer.company is not null` in where clause, it will read full schema of `employ.company`.